### PR TITLE
CORE-7853: Simulator Stubs for alpha release

### PR DESCRIPTION
**Pull request by ashutoshmeher-r3**
This PR adds error messages for unimplemented API's in the Simulator.

When a CorDapp developer tries to use Simulator for an unimplemented API an error message is presented.

### CODE REVIEW

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt [hunk 1]
```diff
@@ -27,3 +35,28 @@ val MemberX500Name.sandboxName: Any
...
+                throw NonImplementedAPIException(it.type.name)
+            }
+        }
+        else
+            throw NonImplementedAPIException("Support for Custom Services")
+    }
+}
+
+private val availableAPIs = listOf(
+    JsonMarshallingService::class.java,
```
Comment (mattbradburyr3) [inline simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:57]: How have we confirmed that these are all the services exposed as part of the alpha API

**Review by LizR3 — DISMISSED** [review]

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt [hunk 2]
```diff
@@ -27,3 +35,28 @@ val MemberX500Name.sandboxName: Any
             .replace(" ", "_")
             .replace(",", "")
     }
+
+fun checkAPIAvailability(flow: Flow){
```
Comment (mattbradburyr3) [inline simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:39]: Do we have instances of services which are partly implemented, ie some of the APIs for the service work but some don't. this method of picking things up wouldn't work for those.

**Review by mattbradburyr3 — DISMISSED** [review]
Approving not to block, but I'd like responses on the comments.

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt (hunk 1 above)
Comment (ashutoshmeher-r3) [reply simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:57]: These are the services we have implemented in Simulator, if there are more that are not implemented in Simulator they should also give the same error response.

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt (hunk 2 above)
Comment (ashutoshmeher-r3) [reply simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:39]: True it won't work for partially implemented APIs but do want to expose partially implemented API. I would believe we don't expose an API in SImulator unless its not fully in sync with Corda

Comment (LizR3) [reply simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:39]: We do implement some partial APIs. @ashutoshmeher-r3 as discussed I think the exception should just be a NotImplementedError, not a custom exception; that would then be consistent with the partially implemented services which automatically get given a `TODO()`.

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt (hunk 1 above)
Comment (LizR3) [reply simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:57]: I like this solution very much as it picks up all missing services dynamically. (The list being tested is the list of our implemented services, not the missing ones!)

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt (hunk 2 above)
Comment (ashutoshmeher-r3) [reply simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/utils/Utilities.kt:39]: Replaced custom exception with NotImplementedError and added SignatureSpecService to available services list

**Review by LizR3 — APPROVED** [review]

**Pull request merged**